src/mlpack/core/data/load_csv.hpp

```diff
@@ -119,33 +205,55 @@ class LoadCSV
     while (std::getline(inFile, line))
     {
       ++rows;
-      // Remove whitespace from either side.
-      boost::trim(line);
-
+      // Remove whitespaces from either side
+      trim(line);
       if (rows == 1)
       {
         // Extract the number of columns.
-        auto findColSize = [&cols](iter_type) { ++cols; };
-        qi::parse(line.begin(), line.end(),
-            stringRule[findColSize] % delimiterRule);
+        std::pair<size_t, size_t> dimen = GetNonNumericMatSize(inFile, delim);
+        cols = dimen.second;
       }
 
       // I guess this is technically a second pass, but that's ok... still the
       // same idea...
       if (MapPolicy::NeedsFirstPass)
       {
         // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
+        std::string str(line.begin(), line.end());
+
+        std::stringstream line_stream;
+        std::string token;
+
+        if(line.size() == 0)
+	{
+	  break;
+        }
+
+        line_stream.clear();
+        line_stream.str(line);
+
+        while(line_stream.good())
         {
-          std::string str(iter.begin(), iter.end());
-          boost::trim(str);
+          std::getline(line_stream, token, delim);
+          // Remove whitespace from either side
+	  trim(token);
+
+          if(token[0] == '"' && token[token.size() - 1] != '"')
```

Comment:
I think some comments here on what the goal of this section is could be really helpful.  I am reading about it and thinking about it and it makes sense, but, it saves the reader/maintainer a lot of time if the goal of the section is clearly denoted in a comment right above the code. :)